quickwit-search/src/fetch_docs.rs

```diff
@@ -150,9 +150,11 @@ async fn fetch_docs_in_split(
     index_storage: Arc<dyn Storage>,
     split: &SplitIdAndFooterOffsets,
 ) -> anyhow::Result<Vec<(GlobalDocAddress, String)>> {
-    let index_reader = get_searcher_for_split(global_doc_addrs.len(), index_storage, split).await?;
+    let num_concurrent_requests = 10;
+    let index_reader = get_searcher_for_split_without_cache(1, index_storage, split).await?;
+    let searcher = Arc::new(index_reader.searcher());
     let doc_futures = global_doc_addrs.into_iter().map(|global_doc_addr| {
-        let searcher = index_reader.searcher();
+        let searcher = searcher.clone();
```

Comment:
oh? So I think we hadn't notice but we accidentally removed the mutability required on the search (I think you introduced a mutex in the search cache a long time ago or something like that).

Doesn't it suggests we can ditch the searcher pool and make the searcher cloneable?